src/Billboard/Billboard.sol

```diff
@@ -47,124 +62,235 @@ contract Billboard is IBillboard {
     }
 
     /// @inheritdoc IBillboard
-    function setBoardName(uint256 tokenId_, string memory name_) external {
-        registry.setBoardName(tokenId_, name_, msg.sender);
+    function setBoardName(uint256 tokenId_, string calldata name_) external isFromBoardCreator(tokenId_) {
+        registry.setBoardName(tokenId_, name_);
     }
 
     /// @inheritdoc IBillboard
-    function setBoardDescription(uint256 tokenId_, string memory description_) external {
-        registry.setBoardDescription(tokenId_, description_, msg.sender);
+    function setBoardDescription(uint256 tokenId_, string calldata description_) external isFromBoardCreator(tokenId_) {
+        registry.setBoardDescription(tokenId_, description_);
     }
 
     /// @inheritdoc IBillboard
-    function setBoardLocation(uint256 tokenId_, string memory location_) external {
-        registry.setBoardLocation(tokenId_, location_, msg.sender);
+    function setBoardLocation(uint256 tokenId_, string calldata location_) external isFromBoardCreator(tokenId_) {
+        registry.setBoardLocation(tokenId_, location_);
     }
 
     /// @inheritdoc IBillboard
-    function setBoardContentURI(uint256 tokenId_, string memory uri_) external {
-        registry.setBoardContentURI(tokenId_, uri_, msg.sender);
+    function setBoardContentURI(uint256 tokenId_, string calldata contentURI_) external isFromBoardTenant(tokenId_) {
+        registry.setBoardContentURI(tokenId_, contentURI_);
     }
 
     /// @inheritdoc IBillboard
-    function setBoardRedirectURI(uint256 tokenId_, string memory redirectURI_) external {
-        registry.setBoardRedirectURI(tokenId_, redirectURI_, msg.sender);
+    function setBoardRedirectURI(uint256 tokenId_, string calldata redirectURI_) external isFromBoardTenant(tokenId_) {
+        registry.setBoardRedirectURI(tokenId_, redirectURI_);
     }
 
     //////////////////////////////
     /// Auction
     //////////////////////////////
 
     /// @inheritdoc IBillboard
-    function setTaxRate(uint256 taxRate_) external isAdmin(msg.sender) {
-        auction.setTaxRate(taxRate_, msg.sender);
+    function getAuction(
+        uint256 tokenId_,
+        uint256 auctionId_
+    ) external view returns (IBillboardRegistry.Auction memory auction) {
+        return registry.boardAuctions[tokenId_][auctionId_];
     }
 
     /// @inheritdoc IBillboard
-    function getTaxRate() external view returns (uint256 taxRate) {
-        return auction.taxRate();
+    function clearAuction(uint256 tokenId_) external {
+        IBillboardRegistry.Board memory board = registry.boards[tokenId_];
+        if (!board) revert BoardNotFound();
+
+        uint256 nextAuctionId = registry.nextBoardAuctionId[tokenId_];
+        if (nextAuctionId == 0) revert AuctionNotFound();
+
+        IBillboardRegistry.Auction memory nextAuction = registry.boardAuctions[tokenId_][nextAuctionId];
+
+        // reclaim ownership to board creator if no auction
+        if (!nextAuction) {
```

Comment:
It seems there will always a auction be there after initialization. When and where we remove it so !nextAuction would be true ?